### ISSUE TITLE
[BUG] Orbit download error - hopefully this is transient.

### ISSUE THREAD

**Issue by cmarshak** [dbekaert/RAiDER#576]
**Describe the bug**

Approximately 1% of GUNW jobs are failing with the below API error at RAiDER step.

Here is an example of inputs and log. Please investigate.

```
{'job_id': 'a897b1f7-0a0d-41ac-b6c7-f3b09644c7e9',
  'job_type': 'INSAR_ISCE_TEST',
  'request_time': '2023-08-17T20:30:41+00:00',
  'status_code': 'FAILED',
  'user_id': 'cmarshak',
  'name': 'track4-n3',
  'job_parameters': {'compute_solid_earth_tide': True,
   'estimate_ionosphere_delay': True,
   'frame_id': 494,
   'granules': ['S1A_IW_SLC__1SDV_20230807T230644_20230807T230711_049776_05FC51_52AC',
    'S1A_IW_SLC__1SDV_20230807T230709_20230807T230736_049776_05FC51_0329'],
   'secondary_granules': ['S1A_IW_SLC__1SDV_20230714T230643_20230714T230710_049426_05F188_8808',
    'S1A_IW_SLC__1SDV_20230714T230708_20230714T230735_049426_05F188_5416'],
   'weather_model': 'HRRR'},
  'logs': ['https://hyp3-tibet-jpl-contentbucket-81rn23hp7ppf.s3.us-west-2.amazonaws.com/a897b1f7-0a0d-41ac-b6c7-f3b09644c7e9/a897b1f7-0a0d-41ac-b6c7-f3b09644c7e9.log'],
  'expiration_time': '2023-09-02T00:00:00+00:00',
  'processing_times': [8714.966, 6.619]}
```

Here is the log reproduced from the link above.

```
â•­â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•®
 â”‚ I'm building Herbie's default config file.      â”‚
 â•°â•¥â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯
 ðŸ‘·ðŸ»â€â™‚ï¸
 â•­â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•®
 â”‚ You're ready to go.                             â”‚
 â”‚ You may edit the config file here:              â”‚
 â”‚ /home/raider/.config/herbie/config.toml         â”‚
 â•°â•¥â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯
 ðŸ‘·ðŸ»â€â™‚ï¸
/opt/conda/envs/RAiDER/lib/python3.10/site-packages/RAiDER/cli/__main__.py:36: DeprecationWarning: Accessing entry points by index is deprecated. Cast to tuple if needed.
  process_entry_point = entry_points(group='console_scripts', name=f'{args.process}.py')[0]
Downloading s3://hyp3-tibet-jpl-contentbucket-81rn23hp7ppf/96473d6f-d41e-4950-a510-7d1d0e53dc75/S1-GUNW-A-R-004-tops-20230807_20230620-230825-00080W_00041N-PP-86ec-v3_0_0.nc to S1-GUNW-A-R-004-tops-20230807_20230620-230825-00080W_00041N-PP-86ec-v3_0_0.nc
Downloading s3://hyp3-tibet-jpl-contentbucket-81rn23hp7ppf/96473d6f-d41e-4950-a510-7d1d0e53dc75/S1-GUNW-A-R-004-tops-20230807_20230620-230825-00080W_00041N-PP-86ec-v3_0_0.json to S1-GUNW-A-R-004-tops-20230807_20230620-230825-00080W_00041N-PP-86ec-v3_0_0.json
Traceback (most recent call last):
  File "/opt/conda/envs/RAiDER/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/conda/envs/RAiDER/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/opt/conda/envs/RAiDER/lib/python3.10/site-packages/RAiDER/cli/__main__.py", line 46, in <module>
    main()
  File "/opt/conda/envs/RAiDER/lib/python3.10/site-packages/RAiDER/cli/__main__.py", line 42, in main
    process_entry_point.load()()
  File "/opt/conda/envs/RAiDER/lib/python3.10/site-packages/RAiDER/cli/raider.py", line 529, in calcDelaysGUNW
    path_cfg, wavelength = RAiDER.aria.prepFromGUNW.main(args)
  File "/opt/conda/envs/RAiDER/lib/python3.10/site-packages/RAiDER/aria/prepFromGUNW.py", line 358, in main
    GUNWObj = GUNW(args.file, args.weather_model, args.output_directory)
  File "<string>", line 6, in __init__
  File "/opt/conda/envs/RAiDER/lib/python3.10/site-packages/RAiDER/aria/prepFromGUNW.py", line 124, in __post_init__
    self.OrbitFile  = self.get_orbit_file()
  File "/opt/conda/envs/RAiDER/lib/python3.10/site-packages/RAiDER/aria/prepFromGUNW.py", line 238, in get_orbit_file
    path_orb = download_eofs([dt], [sat], save_dir=orbit_dir)
  File "/opt/conda/envs/RAiDER/lib/python3.10/site-packages/eof/download.py", line 89, in download_eofs
    query = client.query_orbit_by_dt(orbit_dts, missions, orbit_type=orbit_type)
  File "/opt/conda/envs/RAiDER/lib/python3.10/site-packages/eof/scihubclient.py", line 168, in query_orbit_by_dt
    self._select_orbit(
TypeError: ScihubGnssClient._select_orbit() got an unexpected keyword argument 'orbit_type'
```

@asjohnston-asf and/or @jtherrmann are there reasons that [hyp3lib](https://github.com/ASFHyP3/hyp3-lib) uses both ESA and ASF for the orbit download vs. [sentineleof](https://github.com/scottstanie/sentineleof) just uses ESA.

@bbuzz31 - will likely be good to update prepGUNW accordingly if we can improve robustness via hitting ESA and ASF.

See the hyp3 code - it cycles through both endpoints: https://github.com/ASFHyP3/hyp3-lib/blob/develop/hyp3lib/get_orb.py#L110-L147

We use hyp3lib in dockerized topsapp and cannot remember last time we had issues downloading orbit: https://github.com/ACCESS-Cloud-Based-InSAR/DockerizedTopsApp/blob/dev/isce2_topsapp/localize_orbits.py

**Comment by asjohnston-asf** [comment]
I'm not familiar with the `sentineleof` package or this particular error message.

Years past ASF had seen occasional outages of the Copernicus orbit service. I haven't revisited uptime in a few years but anecdotally it's improved. We saw breaking changes introduced to the service API in 2021 and again in 2022, however, which having a fallback to ASF's orbit archive helped mitigate.

**Comment by asjohnston-asf** [comment]
Looks like a bug in `sentineleof`. `_select_orbit` defined at https://github.com/scottstanie/sentineleof/blob/master/eof/scihubclient.py#L88 has no `orbit_type` parameter, but it's called with that parameter at https://github.com/scottstanie/sentineleof/blob/master/eof/scihubclient.py#L168 .

**Comment by asjohnston-asf** [comment]
At a glance this might only impact recent scenes for which no precision orbit is available. I notice your example is from 2023-08-07. Worth checking if your other failures are from recent dates.

**Comment by cmarshak** [comment]
To be clear though, both libraries provide access to restituted and precise orbits (the former for recent acquisitions). May want to ensure we are using API correctly.

**Comment by scottstanie** [comment]
This was my fault- I added another fix+test, it's up on pypi now with v0.8.1, and i'll bump the conda-forge version too
